package NIznanka;

option optimize_for = SPEED;

message TActionPath {
    optional string Path = 1;
    optional string PathSource = 2;
    optional string PathDestination = 3;
    optional string Action = 4;
}

message TIznankaActionPathCounters {
    repeated float Counters = 1;
    required int64 Timestamp = 2;
    optional float SumWeightedDwellTimes = 3;
    optional float SumWeights = 4;
    optional float SumWeightedDwellTimesPrev = 5;
}

message TIznankaKeyCounters {
    map<string, TIznankaActionPathCounters> ActionPathCounters = 1;
    required int64 MaxReqidTimestamp = 2;
}

message TIznankaPersonalCounters {
    map<string, TIznankaKeyCounters> SourceUrlCounters = 1;
    map<string, TIznankaKeyCounters> FrameUrlCounters = 2;
    required int64 MaxReqidTimestamp = 3;
    map<string, TIznankaKeyCounters> ReqTypeCounters = 4;
}
